runtime/src/chpl-init.c

```diff
@@ -329,6 +329,10 @@ void chpl_std_module_init(void) {
     // Note that in general, module code can contain "on" clauses
     // and should therefore not be called before the call to
     // chpl_comm_startPollingTask().
+    //
+#ifdef HAS_GPU_LOCALE
+    chpl_gpu_enable_device_alloc();
```

Comment:
I wouldn't break up the comment (it seems specific to tasking). 

I'd add a function like `chpl_gpu_on_std_modules_finished_initializing` then get `chpl_gpu_enable_device_alloc` and `chpl_gpu_disable_device_alloc` out of the header for `chpl-gpu` (it'd be better not to give users that hammer).